Automattic/vip-go-mu-plugins
# Prom: speed things up by moving out the rendering logic into a non-WP endpoint

## Description

This PR moves the rendering logic entirely out of WP context. This speeds up the rendering significantly + we're entirely eliminating the issues with sunrise.php in multisite contexts and eliminating any sort of randomness due to the slow customer code. 

The only drawback is right now Post Stats Collector is not working (it requires some core functions to be defined and will need a bit of a refactor)

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

I use a bash function to test:

```bash
function perf {
  curl -o /dev/null -s -w "%{time_connect} + %{time_starttransfer} = %{time_total}\n" "$1"
}
```

Measure the performance locally:

```
perf http://instance.vipdev.lndo.site/.vip-prom-metrics

0.005555 + 0.204087 = 0.204367
```

Apply the PR, and re-run the measurement, you should see an improvement.

```
0.006917 + 0.035189 = 0.035304
```

On VIP, port-forward and do something like 

```
perf http://localhost:8090/.vip-prom-metrics
```
